Transport NSW

Dav0815 opened the pull request:
## Description:
New sensor to request leave times from Transport NSW (Australia). 

**Related issue (if applicable):** fixes #<home-assistant issue number goes here>

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):** home-assistant/home-assistant.io#6529

## Example entry for `configuration.yaml` (if applicable):
```yaml
- platform: transport_nsw
  stop_id: '209516'
  route:  '199'
  api_key: 'YOUR_API_KEY'
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [x] New or updated dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [x] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

Dav0815 committed:
New sensor for Transport NSW

Dav0815 committed:
New sensor for Transport NSW

Dav0815 committed:
New sensor for Transport NSW

Dav0815 committed:
New sensor for Transport NSW

Dav0815 committed:
..

Dav0815 committed:
..

Dav0815 committed:
External lib for requests

Dav0815 committed:
External lib for requests

Dav0815 committed:
New Transport NSW (AU) sensor using PyPi lib

Dav0815 committed:
New Transport NSW (AU) sensor using PyPi lib

Dav0815 committed:
Requirements_all update

Dav0815 committed:
Requirements_all update

Dav0815 committed:
Import fixes and testing

Dav0815 committed:
Import fixes and testing

Dav0815 committed:
Fix of tailing white space

Dav0815 committed:
Fix of tailing white space

gerard33 committed:
Fix a typo (#17147)

lwis committed:
Cleanly stop tradfri on shutdown (#17114)

* Tradfri shutdown fix

* Bump version

* Bump version

* Fix

* Derp

* Remove unnecessary shutdown event

rohankapoorcom committed:
Fix zoneminder zms_url construction (#17150)

Julius2342 committed:
Added warning to HomeKit component (#16807)

* added warning if more then 100 devices are added to HomeKit

cdce8p committed:
Add faucet, shower, sprinkler, valve to HomeKit (#17145)

PaulAnnekov committed:
Fix miflora connection errors during platform setup (#16798)

* possible fix for startup delay

* fixed reported issues

* moved update code into setup

* reverted to previous solution

prophit987 committed:
Discover Danfoss/devolo RS Room Sensor thermostat (#17153)

Allows the Danfoss/Devolo RS Room Sensor thermostat to be discovered as climate device in homeassistant.
This device is defined as GENERIC_TYPE_MULTILEVEL_SENSOR, and not GENERIC_TYPE_THERMOSTAT.
Ref: https://community.home-assistant.io/t/devolo-zwave-thermostat/58739

prophit987 committed:
@danielhiversen as codeowner for Tibber (#17154)

balloob committed:
Update frontend to 20181005.0

balloob committed:
Update translations

balloob committed:
Bumped version to 0.81.0.dev0

balloob committed:
Fix incorrect yaml in hangouts (#17169)

kennedyshead committed:
Adding myself as melissa owner (#17157)

fabaff committed:
Update core, add myself and introduce grouping

fabaff committed:
Revert "Update core, add myself and introduce grouping"

This reverts commit 13106a9b55c0dddb87d5a482ca9a1904911158b1.

balloob committed:
Fix data used for logbook (#17172)

* Fix data used for logbook

* Lint

cdce8p committed:
Fix device_tracker service call & cleanup (#17173)

* Bugfix group service - device_tracker

* Cleanup

fabaff committed:
Update core, add myself and introduce grouping (#17175)

fabaff committed:
Add myself to more sensors (#17185)

amelchio committed:
Upgrade aiolifx_effects to 0.2.1 (#17188)

persandstrom committed:
Verisure standard config for scan interval (#17192)

* verisure configurable polling

* fix indentation

GuyKh committed:
Upgrade boto3 to 1.9.16 (#17140)

cdce8p committed:
Bugfix switch flux - light service call (#17187)

* Bugfix switch flux - light service call

* Change x_val and y_val test

PhilRW committed:
Fix Todoist custom project update (#17115)

* Fix Todoist custom project update

Custom projects were not refreshing the API state and were using
local/stored state. This resulted in invalid tasks being retained upon
update. This change resets the local Todoist API state, syncs it, and
then continues normal update operation(s) on the Todoist project data
object.

* Remove blank line after docstring

* Update logging call

* Simplify logging

persandstrom committed:
vsure version 1.5.0 (#17209)

amelchio committed:
Disallow list/dict for string configuration (#17202)

wonderslug committed:
Enable new registry rename for Insteon (#17171)

* Enable new registry rename for Insteon

* Segment unique_id from name

emontnemery committed:
Fix aliases support for RFLink sensors (#17190)

Danielhiversen committed:
Add forecast for Met.no (#17109)

test met

met no 0.3.0

fix line length

fix line length

Commit:
Add support for ElkM1 alarm/automation panel (#16952)

* Add support for ElkM1 alarm/automation panel.

* fix lines too long

* Address PR comments

* Fix hound ci errors

* Changes for PR comments

* Use vol.Range for checking range value

* Address PR comments

* Fix lint errors

* Added elkm1-lib requirement

* Update coverage to exclude elk

* Fix flake8 errors

* Fix flake8 error

* Cleanup config parsing

* Add housecode converter

* fix PR comments

* fix syntax error

* Fix PR comment

amelchio committed:
Add config entry for LIFX (#17201)

* Add config entry for LIFX

* Use list for dependencies

* Obsolete the platform config

* Use DOMAIN

* Use async_create_task

balloob committed:
Update frontend to 20181007.0

balloob committed:
Update translations

mbrrg committed:
Init sub-components using global var. (#17220)

MartinHjelmare commented on homeassistant/components/sensor/transport_nsw.py:
```diff
@@ -0,0 +1,131 @@
+        _data = self.tnsw.get_departures(self._stopid,
+                                         self._route,
+                                         self._apikey)
+        self.info = [{ATTR_ROUTE: _data[0]['route'],
```
Why create a list with one dictionary and not just a dictionary?

Dav0815 replied:
I plan to add bus after next leaving with an upcoming version.

MartinHjelmare replied:
Unless we add it in this PR, please keep the code clean.

MartinHjelmare commented on homeassistant/components/sensor/transport_nsw.py:
```diff
@@ -0,0 +1,131 @@
+                      ATTR_DUE_IN: _data[0]['due'],
+                      ATTR_DELAY: _data[0]['delay'],
+                      ATTR_REALTIME: _data[0]['realtime']}]
+        return
```
Not needed return.

MartinHjelmare commented on homeassistant/components/sensor/transport_nsw.py:
```diff
@@ -0,0 +1,131 @@
+                      ATTR_REALTIME: 'n/a'}]
+        self.tnsw = TransportNSW.TransportNSW()
+
+    def update(self):
```
Do we need to throttle this to not spam the provider?

Dav0815 replied:
There are no limitations from the provider and getting updates regularly is a key driver for the API from TransportNSW.

MartinHjelmare commented on tests/components/sensor/test_transport_nsw.py:
```diff
@@ -0,0 +1,38 @@
+        """Stop everything that was started."""
+        self.hass.stop()
+
+    def test_transportnsw_config(self):
```
Please mock out the api query and result. We don't want to do any I/O in the tests.

Dav0815 replied:
Fixed

Dav0815 commented on homeassistant/components/sensor/transport_nsw.py:
```diff
@@ -0,0 +1,131 @@
+CONF_ATTRIBUTION = "Data provided by Transport NSW"
+CONF_STOP_ID = 'stopid'
+CONF_ROUTE = 'route'
+CONF_APIKEY = 'apikey'
```
Sure to add the api key for TransportNSW to the const.py ? The key would only be relevant for users in NSW Australia.

MartinHjelmare replied:
I mean that we should import the string constant from const.py.

Dav0815 committed:
Naming and test fixes

Dav0815 committed:
Naming and test fixes

Dav0815 committed:
Lint fixes

Dav0815 committed:
Lint fixes

Dav0815 committed:
Update lib requirements

Dav0815 committed:
Update lib requirements

Danielhiversen committed:
danielhiversen as codeowner for met.no (#17232)

Dav0815 committed:
Mock test and requirements

Dav0815 committed:
Mock test and requirements

balloob committed:
Prevent accidental device reg override (#17136)

syssi committed:
Add some new model names of Xiaomi Aqara devices (#17234)

* Add additional model name of the Xiaomi Aqara Button (WXKG11LM)

* Add additional model name of the Xiaomi Aqara Wireless Switch (WXKG02LM, WXKG03LM)

* Bump PyXiaomiGateway version

Commit:
Pushed to version 0.7.6 of denonavr library to add more sound modes (#17227)

MartinHjelmare commented on homeassistant/components/sensor/transport_nsw.py:
```diff
@@ -0,0 +1,130 @@
+CONF_ATTRIBUTION = "Data provided by Transport NSW"
+CONF_STOP_ID = 'stop_id'
+CONF_ROUTE = 'route'
+CONF_API_KEY = 'api_key'
```
Since we import this from const.py we can remove the definition here.

MartinHjelmare commented on tests/components/sensor/test_transport_nsw.py:
```diff
@@ -0,0 +1,53 @@
+    def setUp(self):
+        """Set up things to run when tests begin."""
+        self.hass = get_test_home_assistant()
+        self.config = VALID_CONFIG
```
We don't need to store this as an instance attribute. `VALID_CONFIG` is a constant, so we can use it in the whole module.

MartinHjelmare commented on tests/components/sensor/test_transport_nsw.py:
```diff
@@ -0,0 +1,53 @@
+        """Set up things to run when tests begin."""
+        self.hass = get_test_home_assistant()
+        self.config = VALID_CONFIG
+        self.reference = {}
```
This can be remove. It's not used.

MartinHjelmare commented on tests/components/sensor/test_transport_nsw.py:
```diff
@@ -0,0 +1,53 @@
+        self.hass = get_test_home_assistant()
+        self.config = VALID_CONFIG
+        self.reference = {}
+        self.entities = []
```
This can be remove. It's not used.

MartinHjelmare reviewed (commented):
Looks good! Some minor comments left.

balloob committed:
Fix SPC (#17236)

syssi committed:
Fix data_key and power_consumed attribute of the Aqara Wall Switch (Closes: #16457) (#17235)

Commit:
Water heater support (#17058)

* Moved econet to water_heater

* Wink and Econet are now in water heater!

* Removed away mode from econet and added demo water heater

* Added demo tests

* Updated coveragerc

* Fix lint issues.

* updated requirements all

* Requirements all actually updated.

* Reset wink and econet and fixed service.

* Reset wink and econet to the correct dev state

* Reset requirements_all and .coveragerc and removed the new econet and wink water_heater files

* Removed @bind_hass service methods

* Actually reset the .coverage file

* Fixed the tests

* Addressed @MartinHjelmare's comments

* Removed unused import

* Switched to async_add_executor_job

* Fixed lint

* Removed is_on

* Added celsius demo water heater and tests.

* Removed metric import

MatteGary committed:
Daikin Climate - Better integration with Climate base component (#16913)

* Daikin Climate - Better integration with Climate base component

Made some modification in order to better integrate the Daikin AC Component with the base Climate Component.
Benefits are:
Support localization for Operation Mode
Support for Homekit Integration (if the AC is turned On, now the status is updated in Homekit)

* Build bug fixing

* Bug fixing in Set Operation_Mode functionality

Fixed to .title() functionality in matching the Operation_Mode

* Fix useless code

* Fix hound bug

* Bug fixing

Change in list of Operation Mode

* Trailing white space fix

* Compile fixing

* Whitespace fix

Dav0815 committed:
Clean up

OttoWinter committed:
Fix potential MQTT discovery race condition (#17208)

* Fix potential MQTT discovery race condition

* Rename data key

Dav0815 committed:
Merge conflict fix

homeassistant commented:
Hi @prophit987, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#17152).

Once you do that we will be able to review and accept this pull request.

Thanks!

Dav0815 committed:
Merge conflict fix

homeassistant commented:
Hi @prophit987, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#17152).

Once you do that we will be able to review and accept this pull request.

Thanks!
